src/index.ts

```diff
@@ -1,17 +1,115 @@
+import { getDataStreamDcatAp201 } from './dcat-ap';
+import * as config from 'config';
 
 export = class Output {
-    static type = 'output';
-    static version = version;
-    static routes = [
-      {
-        path: '/content',
-        methods: ['get'],
-        handler: 'serve'
+  static type = 'output';
+  static version = version;
+  static routes = [
+    {
+      path: '/dcat-ap/2.0.1',
+      methods: ['get'],
+      handler: 'serve',
+    },
+  ];
+
+  model: any;
+
+  public async serve (req: Request, res: Response) {
+    res.set('Content-Type', 'application/json');
+
+    try {
+      const { domainRecord, siteModel } = await this.fetchDomainAndSite(req.hostname);
+
+      const siteCatalog = _.get(siteModel, 'data.catalog');
+      if (!siteCatalog) {
+        res.status(200).send({});
+      } else {
+       const dcatStream = getDataStreamDcatAp201({
+        domainRecord,
+        siteItem: siteModel.item,
+        env: this.getEnvFromPortal(this.portalUrl),
+      });
+
+      req.res.locals.searchRequest = this.getSearchRequestFromCatalog(siteCatalog, this.portalUrl);
+
+      const datasetStream = await this.model.pullStream(req);
+
+      datasetStream
+        .pipe(dcatStream)
+        .pipe(res)
+        .on('error', (err: any) => {
+          res.status(500).send(this.getErrorResponse(err));
+        });
+      }
+    } catch (err) {
+      res.status(500).send(this.getErrorResponse(err));
+    }
+  }
+
+  private async fetchDomainAndSite (hostname) {
+    const requestOptions = this.getRequestOptions(this.portalUrl);
+
+    const domainRecord = (await lookupDomain(
+      hostname,
+      requestOptions,
+    )) as IDomainEntry;
+    const siteModel = await getSiteById(domainRecord.siteId, requestOptions);
+
+    return { domainRecord, siteModel };
+  }
+
+  private getRequestOptions (portalUrl: string): IHubRequestOptions {
+    return {
+      isPortal: false,
+      hubApiUrl: getHubApiUrl(portalUrl),
+      portal: getPortalApiUrl(portalUrl),
+      authentication: null,
+    };
+  }
+
+  private get portalUrl () {
+    return config.has('arcgisPortal')
+      ? config.get('arcgisPortal') as string
+      : 'https://www.arcgis.com';
+  }
+
+  private getSearchRequestFromCatalog (catalog: any, portalUrl: string): IContentSearchRequest {
+     return {
+      filter: {
+        group: catalog.groups,
+        orgid: catalog.orgId,
+      },
+      options: {
+        portal: portalUrl
       }
-    ];
+    };
+  }
 
-    // async serve (req, res) {
+  private getErrorResponse (err: any) {
+    return { error: _.get(err, 'message', 'Encountered error while processing request') };
+  }
 
-    // }
-}
\ No newline at end of file
+  private getEnvFromPortal (portalUrl: string) {
+    let env;
+    if (/devext\.|mapsdev\./.test(portalUrl)) {
+      env = 'dev';
+    } else if (/qaext\.|mapsqa\./.test(portalUrl)) {
+      env = 'qa';
+    } else {
+      env = 'prod';
+    }
+    return env;
+  }
```

Comment:
Similar to the previous comment, why have this as a class method that gets executed every time `serve` is executed?  it appears that the `env` doesn't change with each request.  So can't you just determine `env` once when the module loads?